modules/ipp/scanner.go

```diff
@@ -0,0 +1,206 @@
+// TODO: Add more protocol-specific flags as needed
+// Flags holds the command-line configuration for the ipp scan module.
+// Populated by the framework.
+type Flags struct {
+	zgrab2.BaseFlags
+	//FIXME: Borrowed from http module
+	MaxRead int `long:"max-size" default:"256" description:"Max kilobytes to read in response to an HTTP request"`
+	// TODO: Protocols that support TLS should include zgrab2.TLSFlags (do once implemented)
+	// TODO: Maybe implement both an ipps connection and upgrade to https
+	IPPSecure bool `long:"ipps" description:"Perform a TLS handshake immediately upon connecting."`
+
+	Verbose bool `long:"verbose" description:"More verbose logging, include debug fields in the scan results"`
+}
+
+// Module implements the zgrab2.Module interface.
+type Module struct {
+	// TODO: Add any module-global state if necessary
+}
+
+// Scanner implements the zgrab2.Scanner interface.
+type Scanner struct {
+	config *Flags
+	// TODO: Add scan state if any is necessary
+}
+
+// RegisterModule registers the zgrab2 module.
+func RegisterModule() {
+	var module Module
+	_, err := zgrab2.AddCommand("ipp", "ipp", "Probe for ipp", 631, &module)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+// NewFlags returns a default Flags object.
+func (module *Module) NewFlags() interface{} {
+	return new(Flags)
+}
+
+// NewScanner returns a new Scanner instance.
+func (module *Module) NewScanner() zgrab2.Scanner {
+	return new(Scanner)
+}
+
+// Validate checks that the flags are valid.
+// On success, returns nil.
+// On failure, returns an error instance describing the error.
+func (flags *Flags) Validate(args []string) error {
+	return nil
+}
+
+// Help returns the module's help string.
+func (flags *Flags) Help() string {
+	//TODO: Write a help string
+	return ""
+}
+
+// Init initializes the Scanner.
+func (scanner *Scanner) Init(flags zgrab2.ScanFlags) error {
+	f, _ := flags.(*Flags)
+	scanner.config = f
+	//TODO: Take action in response to flags which were set
+	return nil
+}
+
+// InitPerSender initializes the scanner for a given sender.
+func (scanner *Scanner) InitPerSender(senderID int) error {
+	return nil
+}
+
+// GetName returns the Scanner name defined in the Flags.
+func (scanner *Scanner) GetName() string {
+	return scanner.config.Name
+}
+
+// Protocol returns the protocol identifier of the scan.
+func (scanner *Scanner) Protocol() string {
+	return "ipp"
+}
+
+// GetPort returns the port being scanned.
+func (scanner *Scanner) GetPort() uint {
+	return scanner.config.Port
+}
+
+//FIXME: Maybe switch to ipp/ipps schemes, at least optionally
+func getIPPURL(https bool, host string, port uint16, endpoint string) string {
+	var proto string
+	if https {
+		proto = "https"
+	} else {
+		proto = "http"
+	}
+	return proto + "://" + host + ":" + strconv.FormatUint(uint64(port), 10) + endpoint
+}
+
+func ippInContentType(resp http.Response) bool {
+	for _, t := range resp.Header["Content-Type"] {
+		if strings.Contains(t, CONTENT_TYPE) {
```

Comment:
This is really nit-picky, but this would match e.g. `"application/ippimississ"`; there should be a library for checking content types, but if not, something like `t == CONTENT_TYPE or strings.StartsWith(t, CONTENT_TYPE + ";")` with a reference to the RFC would do in a pinch.